### PR TITLE
[DOCS] Updating dynamic mappings for runtime fields

### PR DESCRIPTION
**Pull request by lockewritesdocs**
Updates the dynamic templates page with more information about runtime fields.

Preview link: https://elasticsearch_67301.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/dynamic-templates.html

### CODE REVIEW

File: docs/reference/mapping/dynamic/templates.asciidoc [hunk 1]
```diff
@@ -62,6 +65,47 @@ putting new dynamic templates through the <<indices-put-mapping, put mapping>> A
...
+PUT my-index-000001/
+{
+  "mappings": {
+    "dynamic_templates": [
+      {
+        "strings_as_ip": {
+          "match_mapping_type": "string",
+          "match": "ip*",
+          "runtime": {
+            "type": "ip"
```
Comment (javanna) [inline docs/reference/mapping/dynamic/templates.asciidoc:97]: great example, thanks!

File: docs/reference/mapping/dynamic/templates.asciidoc [hunk 2]
```diff
@@ -62,6 +65,47 @@ putting new dynamic templates through the <<indices-put-mapping, put mapping>> A
...
+If you want {es} to dynamically map new fields of a certain type as runtime
+fields, set `dynamic:runtime` in the index mappings. These fields are not
+indexed, and are loaded from `_source` at query time.
+
+Alternatively, you can use the default dynamic mapping rules and then create
+dynamic templates to map specific fields as runtime fields. You set
+`dynamic:true` in your index mapping, and then create a dynamic template to map
+new fields of a certain type as runtime fields.
+
+Let's say you have data where each of the fields start with `ip_`. Based on the
```
Comment (javanna) [inline docs/reference/mapping/dynamic/templates.asciidoc:79]: shall we specify here that this example applies to both dynamic:true or dynamic:runtime?

**Review by javanna — APPROVED** [review]
left a small comment, LGTM otherwise. Thanks!

File: docs/reference/mapping/dynamic/templates.asciidoc (hunk 2 above)
Comment (lockewritesdocs) [reply docs/reference/mapping/dynamic/templates.asciidoc:79]: Good call. I think we can split the explanation into two paragraphs like this:

>Let's say you have data where each of the fields start with `ip_`. Based on the
<<match-mapping-type,dynamic mapping rules>>, {es} maps any `string` that passes
`numeric` detection as a `float` or `long`. However, you can create a dynamic
template that maps new strings as runtime fields of type `ip`.

>The following request defines a dynamic template named `strings_as_ip`. When
{es} detects new `string` fields matching the `ip*` pattern, it maps those
fields as runtime fields of type `ip`. You can use this template with either
`dynamic:true` or `dynamic:runtime`.

Comment (javanna) [reply docs/reference/mapping/dynamic/templates.asciidoc:79]: Last sentence, shall we adapt it to something like "You can use this template with either dynamic:true or dynamic:runtime, as ip fields don't get dynamically mapped with either of the two". I am sure you will prettify it but that would be the message ;)

**Pull request merged**